### ISSUE TITLE
Suggests return typehint from phpdoc

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#256]
Following #28 too, it should suggests the typehint from phpdoc and vice-versa. For instance: if I declare a singular type or nullable on phpdoc (eg. `string` or `?string` for _7.1_), then suggests to apply on other side.

```php
/** @return string */
public function test() {
    // Warning: you can declare return typehint of function test() to string.
    return '';
}
```

```php
// Warning: you can declare phpdoc return typehint of function test() to string.
public function test(): string {
    return '';
}
```

For PHP 7.1, you should accepts nullable to: `?string` or some else. For complex types, this inspection should be ignored.

**Comment by kalessil** [comment]
Not sure why this ticket is needed - code samples have no return statements, hence PhpDoc is totally incorrect.

@rentalhost : did you try extending cases with valid returns - the inspection expected to report as you want then.

**Comment by rentalhost** [comment]
Edited,

But basically, I mean to sync phpdoc with php typehint.

**Comment by Fleshgrinder** [comment]
Why would we need phpDoc if we have a return type?

**Comment by rentalhost** [comment]
Yeah, it make sense. Then maybe it could do the inverse? Check if both are declared then warn about that (QF transfer phpdoc to return typehint, case is omited).

**Comment by Fleshgrinder** [comment]
This is a very problematic situation. One the one hand we do not want to duplicated things, since it is a waste of effort. On the other hand we need PhpDoc to define extended return type information. Consider again the delegate example I gave in another issue:

```php
class Delegate {
    public function __invoke() {}
}

/** @return callable|Delegate */
function f(): Delegate {}
```

The additional `callable` is required because PhpStorm will not understand that an instance of `Delegate` is actually a _callable_ just because it implements the `__invoke` method. There are many more situations with iterators where we might want to have a more specific type:

```php
class Iter extends IteratorAggregate {
    /** @return Delegate[]|\Generator */
    public function getIterator(): \Generator {}
}
```

**Comment by rentalhost** [comment]
Closing: is better avoid duplicates. We should prioritize the php typehint declaration.

**Comment by kalessil** [comment]
Good decision on closing this, thank you.
